tests/Backend/Bigquery/ImportTreatValuesAsNullTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Keboola\Test\Backend\Bigquery;
+
+use Generator;
+use Keboola\Csv\CsvFile;
+use Keboola\StorageApi\Workspaces;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+use Keboola\Test\Backend\Workspaces\ParallelWorkspacesTestCase;
+use PDO;
+
+class ImportTreatValuesAsNullTest extends ParallelWorkspacesTestCase
+{
+    /**
+     * @dataProvider treatValuesAsNullData
+     */
+    public function testImportTreatValuesAsNull(
+        string $importedFile,
+        array $expectedData,
+        ?array $treatValuesAsNull = null
+    ): void {
+        $this->allowTestForBackendsOnly([
+            self::BACKEND_BIGQUERY,
+        ]);
+        $tableName = 'testImportTreatValuesAsNull';
+        $tableId = $this->createTable($tableName);
+
+        $params = [];
+        if ($treatValuesAsNull !== null) {
+            $params = ['treatValuesAsNull' => $treatValuesAsNull,];
+        }
+        $this->_client->writeTableAsync($tableId, new CsvFile($importedFile), $params);
+
+        $workspaces = new Workspaces($this->workspaceSapiClient);
+        $workspace = $workspaces->createWorkspace([], true);
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $options = [
+            'input' => [
+                [
+                    'source' => $tableId,
+                    'destination' => $tableName,
+                    'useView' => true,
+                ],
+            ],
+            'preserve' => false,
+        ];
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
+        $data = $backend->fetchAll($tableName, PDO::FETCH_NUM, 'Id');
+        $data = array_map(function ($row) {
+            unset($row[4]); // remove timestamp
+            return $row;
+        }, $data);
+
+        $this->assertTrue($data === $expectedData);
```

Comment:
Toho sem si všiml až teď co je to sakra za assert? :D Proč ne assertSame nebo assertEqualsCanonicalizing?

Reply:
No jo, pekna kravina...